genie-web/src/main/java/com/netflix/genie/web/data/entities/CommandEntity.java

```diff
@@ -274,6 +289,68 @@ protected void setClusters(@Nullable final Set<ClusterEntity> clusters) {
         }
     }
 
+    /**
+     * Set the criteria, in priority order, that this command has for determining which available clusters to use
+     * for a job.
+     *
+     * @param clusterCriteria The cluster criteria
+     */
+    public void setClusterCriteria(@Nullable final List<CriterionEntity> clusterCriteria) {
+        this.clusterCriteria.clear();
+        if (clusterCriteria != null) {
+            this.clusterCriteria.addAll(clusterCriteria);
+        }
+    }
+
+    /**
+     * Add a new cluster criterion as the lowest priority criterion to evaluate for this command.
+     *
+     * @param criterion The {@link CriterionEntity} to add
+     */
+    public void addClusterCriterion(final CriterionEntity criterion) {
+        this.clusterCriteria.add(criterion);
+    }
+
+    /**
+     * Add a new cluster criterion with the given priority.
+     *
+     * @param criterion The new criterion to add
+     * @param priority  The priority with which this criterion should be considered. {@literal 0} would be the highest
+     *                  priority and anything greater than the current size of the existing criteria will just be added
+     *                  to the end of the list. If a priority of {@code < 0} is passed in the criterion is added
+     *                  as the highest priority ({@literal 0}).
+     */
+    public void addClusterCriterion(final CriterionEntity criterion, final int priority) {
+        if (priority <= HIGHEST_CRITERION_PRIORITY) {
```

Comment:
This only works if priority is 0, and throws if it's negative.
Maybe the condition should be `< size`?

Reply:
Not sure I follow what you're saying here. Unless I'm missing something the constant `HIGHEST_CRITERION_PRIORITY` is `0` so this is `<=0` it will just use 0 so it won't throw exception. If it's `>=` the current size it appends it at the end. any other case it inserts it. Do you think this is wrong?